doc/md/fundamentals/2-types/10-options-results.md

```diff
@@ -83,7 +89,6 @@ assert get(?"A", "B") == "A";
 assert get(null, "B") == "B";
 ```
 
```

Comment:
```suggestion
The `let` matches option against the pattern `?value`, extracting any contents from `option`. If `option` is `null`, the pattern fails to match, and the `else` branch is entered, exiting the function with `defaultValue`.
```

Using a switch, we could have implemented this as:

``` motoko
func get<T>(option : ?T, defaultValue : T) : T {
  switch option {
    case null defaultValue;
    case (?value) value;
  }
};
```
This is a little longer but also increases the nesting of the code that needs to use `value`.

::: tip

Although convenient for option patterns, `let-else` also works with other types of pattern,
handling pattern-match failure using the `else` branch.

:::